src/api/db/models/Image.js

```diff
@@ -316,6 +322,82 @@ export class ImageModel {
     }
   }
 
+  static async deleteComment(input, context) {
+    try {
+      const image = await ImageModel.queryById(input.imageId, context);
+
+      const comment = (image.comments || []).filter((c) => { return c._id === input.id; })[0];
+      if (!comment) throw new Error('Comment not found on image');
+
+      if (comment.author !== context.user['cognito:username'] && !context.user['is_superuser']) {
+        throw new Error('Can only edit your own comments');
+      }
+
+      image.comments = image.comments.filter((c) => { return c._id !== input.id; });
+
+      await image.save();
+
+      return { message: 'Images Deleted' };
```

Comment:
@ingalls I just mentioned this in Slack, but I think it would be a bit cleaner on the front end if all of these mutations returned the updated `image.comments` array.